mapreduce/src/main/java/uk/gov/gchq/palisade/PalisadeInputSplit.java

```diff
@@ -0,0 +1,177 @@
+     * @param resources the resources to be processed by this split
+     * @throws NullPointerException if anything is null
+     */
+    public PalisadeInputSplit(final RequestId requestId, final Map<Resource, ConnectionDetail> resources) {
+        Objects.requireNonNull(requestId, "requestId");
+        Objects.requireNonNull(resources, "resources");
+        requestResponse = new DataRequestResponse(requestId, resources);
+    }
+
+    /**
+     * Create a new input split. The request response is stored inside this input split.
+     *
+     * @param requestResponse the response object
+     * @throws NullPointerException if {@code requestResponse} is null
+     */
+    public PalisadeInputSplit(final DataRequestResponse requestResponse) {
+        Objects.requireNonNull(requestResponse);
+        this.requestResponse = new DataRequestResponse(requestResponse);
+    }
+
+    /**
+     * Get the response for this input split. This response object may contain many resources for a single request.
+     *
+     * @return the response object
+     */
+    public DataRequestResponse getRequestResponse() {
+        return requestResponse;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * We measure length according to the number of resources that this split will process.
+     *
+     * @return the number of resources contained in this split
+     */
+    @Override
+    @JsonIgnore
+    public long getLength() throws IOException, InterruptedException {
+        return getRequestResponse().getResources().size();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * We don't implement about data locality, so this always returns an empty array.
+     *
+     * @return always returns an empty string array
+     */
+    @Override
+    @JsonIgnore
+    public String[] getLocations() throws IOException, InterruptedException {
+        return StringUtils.emptyStringArray;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void write(final DataOutput dataOutput) throws IOException {
+        Objects.requireNonNull(dataOutput, "dataOutput");
+        //serialise this class to JSON and write out
+        byte[] serial = JSONSerialiser.serialise(this);
+        dataOutput.writeInt(serial.length);
+        dataOutput.write(serial);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void readFields(final DataInput dataInput) throws IOException {
+        Objects.requireNonNull(dataInput, "dataInput");
+        int length = dataInput.readInt();
+        //validate length
+        if (length < 0) {
+            throw new IOException("illegal negative length on deserialisation");
+        }
+        //make buffer and read
+        byte[] buffer = new byte[length];
+        dataInput.readFully(buffer);
+        //deserialise
+        PalisadeInputSplit other = JSONSerialiser.deserialise(buffer, PalisadeInputSplit.class);
+        Objects.requireNonNull(other.requestResponse);
```

Comment:
Could you add an error message as a second argument to Objects.requireNonNull please?

Reply:
Done